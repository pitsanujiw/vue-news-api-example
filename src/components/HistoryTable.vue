<template>
  <v-simple-table dark>
    <template v-slot:default>
      <thead>
        <tr>
          <td>Title</td>
          <td>Source</td>
          <td>Autdor</td>
          <td>Visitor dates</td>
        </tr>
      </thead>
      <tbody>
        <tr v-for="(item, key) in getHistories" :key="key">
          <td>{{ item.history.title }}</td>
          <td>{{ item.history.source.name }}</td>
          <td>{{ item.history.author || '-' }}</td>
          <td>{{ onChangeFormatDate(item.date) }}</td>
        </tr>
      </tbody>
    </template>
  </v-simple-table>
</template>

<script>
import { mapGetters } from 'vuex';

import { onChangeFormatDate } from '@/utils';

export default {
  name: 'HistoryTable',
  data: () => ({}),
  methods: {
    onChangeFormatDate,
  },
  computed: {
    ...mapGetters(['getHistories']),
  },
};
</script>

<style>
</style>
