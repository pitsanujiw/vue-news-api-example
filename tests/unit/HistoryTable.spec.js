import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';

import HistoryTable from '../../src/components/HistoryTable.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(Vuetify);

describe('NewsCard', () => {
  const modules = {
    getters: {},
  };
  let store;

  beforeEach(() => {
    jest.clearAllMocks();
    store = new Vuex.Store({
      modules: {
        newModules: modules,
      },
    });
  });

  it('Renders HistoryTable Component', () => {
    const wrapper = shallowMount(HistoryTable, {
      store,
      localVue,
      vuetify,
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
