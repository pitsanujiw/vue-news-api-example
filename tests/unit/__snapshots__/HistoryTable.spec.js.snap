// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsCard Renders HistoryTable Component 1`] = `
<v-simple-table-stub dark="true">
  <thead>
    <tr>
      <td>Title</td>
      <td>Source</td>
      <td>Autdor</td>
      <td>Visitor dates</td>
    </tr>
  </thead>
  <tbody></tbody>
</v-simple-table-stub>
`;
